### DISCUSSION TITLE
Deploy a PostgreSQL database via Helm

### DISCUSSION THREAD

**Discussion by olimination** [appuio/appuio-cloud-community#27] [Help Needed] [answered]
Hello Appuio Cloud community!
I'm trying to install a PostgreSQL database on the the Appuio Cloud LGP-2 Zone with Helm using this Helm chart: https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml and using the following command:

`helm install -f postgresql-values.yaml --set global.postgresql.postgresqlPassword=MY_PW content-service-db bitnami/postgresql`

The installation was unfortunately not yet successful, I receive following message in the OpenShift console: 
`create Pod content-service-db-postgresql-0 in StatefulSet content-service-db-postgresql failed error: Pod "content-service-db-postgresql-0" is invalid: [spec.containers[0].resources.requests: Invalid value: "256Mi": must be less than or equal to memory limit, spec.containers[0].resources.requests: Invalid value: "250m": must be less than or equal to cpu limit]`. 

If I check the default quotas this should actually be ok and valid or do I misunderstand something?

Also I did not use the _gluster-database_ storageClass as documented here: https://docs.appuio.ch/en/latest/app/helmcharts.html, I removed the _storageClass_ confiugration because this is not available anymore on the new Appuio Cloud platform as far as I understand. But I'm also not sure if I need to configure some other storageClass like "ssd"?
Or is it enough to leave it simply "undefined" and some default storage will be picked up? It seems so that if I leave this storageClass value empty it will pick up the _ssd_ storageClass.

I have used the following values.yaml file for customizing the Helm chart:
```
global:
  postgresql:
    postgresqlUsername: my-user

image:
  tag: 13.5.0-debian-10-r30

persistence:
  size: 1Gi

volumePermissions:
  enabled: false
  securityContext:
    runAsUser: "auto"

securityContext:
  enabled: false

shmVolume:
  chmod:
    enabled: false

containerSecurityContext:
  enabled: false

```

Do you have any hints on how I can run a PostgreSQL database on the new Appuio Cloud platform?
Thanks for support!

**Comment by simu** [comment, marked as answer]
Hi,

> The installation was unfortunately not yet successful, I receive following message in the OpenShift console:
> `create Pod content-service-db-postgresql-0 in StatefulSet content-service-db-postgresql failed error: Pod "content-service-db-postgresql-0" is invalid: [spec.containers[0].resources.requests: Invalid value: "256Mi": must be less than or equal to memory limit, spec.containers[0].resources.requests: Invalid value: "250m": must be less than or equal to cpu limit]`.

I had a quick look and realized that the cause for the error is that the Helm chart only sets resource requests.

However, on APPUiO Cloud we require that users to also configure resource limits if they configure resource requests. If users don't explicitly configure limits, the platform injects limits of `cpu: 200m` and `memory: 200Mi` (see https://docs.appuio.cloud/user/references/default-quota.html#_resource_limits_and_defaults).

In your case, Kubernetes then rejects the configuration since the default requests configured in the Helm chart are larger than the default limits injected by APPUiO Cloud.

I think you should be able to fix this error by adding the following to your `values.yaml`:

```
resources:
  limits:
    cpu: 500m
    memory: 512Mi
``` 




> Also I did not use the _gluster-database_ storageClass as documented here: https://docs.appuio.ch/en/latest/app/helmcharts.html, I removed the _storageClass_ confiugration because this is not available anymore on the new Appuio Cloud platform as far as I understand. But I'm also not sure if I need to configure some other storageClass like "ssd"?
> Or is it enough to leave it simply "undefined" and some default storage will be picked up? It seems so that if I leave this storageClass value empty it will pick up the _ssd_ storageClass.

Yes, on APPUiO Cloud we're not offering Gluster volumes for persistent storage anymore. Instead, we offer a number of storageclasses (RWO and RWX) for different usecases, see https://docs.appuio.cloud/user/explanation/storage-classes.html#_cloudscale_ch for details.

As you've noticed the `ssd` storageclass is configured as the default and will be used for PVCs which don't explicitly request a storageclass. We recommend using the `ssd` storageclass unless you have specific needs which another documented storageclass meets better.

**Reply by olimination** [reply]
Thank you @simu! It worked :+1:
